src/main/java/com/sphy/hotelmanagementapplication/service/RoomService.java

```diff
@@ -177,11 +184,14 @@ public RoomDTO updateRoom(RoomDTO roomDTO) throws NullPointerException{
             existingRoom.setPrice(roomDTO.getPrice());
             hotel.ifPresent(existingRoom::setHotel);
 			existingRoom.setDisabled(roomDTO.isDisabled());
-            roomRepository.save(existingRoom);
+
             hotel.get().getRooms().add(existingRoom);
             hotelRepository.save(hotel.get());
+
+            return roomToRoomDTO.converter(roomRepository.save(existingRoom));
         }
-        return roomDTO;
+        return null;
```

Comment:
I think, the code will have more quality if we wrapped this in an optional - what do you think? Its is not mandatory to change it if you dont want to.

Reply:
it is with optional but if it is not exists it returns null